src/Compilers/Core/Portable/Diagnostic/TriggerDiagnosticDescriptor.cs

```diff
@@ -18,7 +18,7 @@ public class TriggerDiagnosticDescriptor : DiagnosticDescriptor
         /// <param name="customTags">Optional custom tags for the diagnostic. See <see cref="WellKnownDiagnosticTags"/> for some well known tags.</param>
         public TriggerDiagnosticDescriptor(string id, params string[] customTags)
             : base(id, title: "", messageFormat: "", category: "", defaultSeverity: DiagnosticSeverity.Hidden, isEnabledByDefault: true,
-                    customTags: customTags.Append(WellKnownDiagnosticTags.NotConfigurable))
+                    customTags: customTags.Append(WellKnownDiagnosticTags.NotConfigurable).Append(WellKnownDiagnosticTags.Trigger))
```

Comment:
I think is a bit strange that some (in this case one) special descriptors are allowed to round trip - while other ones aren't. What we are telling users is that if you happen to use TriggerDiagnosticDescriptor then you can rely on the type being preserved at the other end - but for other ones we won't.

Isn't it better to say that we never round trip the type (so you would never see the type preserved regardless of what type of descriptor you use)? Wouldn't that be a better way to tell users that if they want to pass data across, they should only rely on tags (and never on the type of the thing being preserved)?